src/store/messagesStore.js

```diff
@@ -290,7 +290,14 @@ const mutations = {
 		if (!state.messages[token]) {
 			Vue.set(state.messages, token, {})
 		}
-		Vue.set(state.messages[token], message.id, message)
+		// TODO split adding and updating message in the store to different actions
+		// message.parent doesn't contain grand-parent, so we should keep it
+		// when updating message in store from new message.parent object
+		const storedMessage = state.messages[token][message.id]
+		const preparedMessage = !message.parent && storedMessage?.parent
+			? { ...message, parent: storedMessage.parent }
+			: message
```

Comment:
let's revert it back ? this just solves if parent existed or not, earlier it could solve any attr missing in parent of a system message.

Reply:
Then we would have "edited" deleted message, because we didn't cleanup lastEdit properties.
Best would be to check with API team, whether we're always passing  all properties apart from parent, or not

Reply:
or if it is possible to add that parent property to the message (without its grand parents if any :p)

Reply:
Only nested parents are ignored (to prevent infinite loop, I guess) 
https://github.com/nextcloud/spreed/blob/1ccbf67da231c37801c1406b10000144125aec35/lib/Controller/ChatController.php#L551-L552
So should be good with current changes?

Reply:
We won't have infinite loop, if we only add parent up to second level.